radis/lbl/calc.py

```diff
@@ -461,9 +461,17 @@ def _check_molecules_are_consistent(
         for argument_name, argument_dict in DICT_INPUT_ARGUMENTS.items():
             if isinstance(argument_dict, dict):
                 # Choose the correspond value
-                molecule_dict[molecule][argument_name] = argument_dict[molecule]
-                # Will raise a KeyError if not defined. That's fine!
-                # TODO: maybe need to catch KeyError and raise a better error message?
+                try:
+                    molecule_dict[molecule][argument_name] = argument_dict[molecule]
+                except KeyError:
+                    # KeyError is raised if a molecule is missing from the dictionary
+                    raise KeyError(
+                        "'{0}'' missing in '{1}=' argument. For reference, here are the molecules provided in the following arguments: 'molecule=' {2} and '{1}=' {3}".format(
+                            molecule,
+                            argument_name,
+                            molecule_reference_set,
+                            set(argument_dict.keys()))
```

Comment:
Thanks for contributing. 

The `f""` syntax is more readable than `"".format()`; can you use it ? I.e `f"'{molecule}' missing in '{argument_name}=' argument (...) `

Can you also provide an example of how it would look like in real-error ?

Reply:
I was working to set up an example error and I could use some clarification on the TODO. I think this TODO may already taken care of earlier in the calc_spectrum function. 

I understood the comment as saying a KeyError occurs in the following line: `molecule_dict[molecule][argument_name] = argument_dict[molecule]`. Specifically because of the `argument_dict[molecule]` part. For example, if the isotope= argument was provided a dict missing a molecule that was provided in molecule=.

Any inconsistencies seem to be pretty well covered by the `_check_molecules_are_consistent `function defined earlier in calc_spectrum, starting on line 402 at the moment. I don't see a line where any of the dictionaries could be modified between the `_check_molecules_are_consistent `function and the line were the TODO was listed.

I think the TODO can be removed and my pull request dropped, but let me know if I missed something.

Reply:
Ok, sounds good ! Then you can adjust your PR to at least remove the TODO , and we'll merge it